### DISCUSSION TITLE
UMPV Questions

### DISCUSSION THREAD

**Discussion by advice2020** [mpv-player/mpv#14490] [Q&A]
Linux Debian based distro / GTK based Desktop Environment
MPV  0.35.1-4

I have recently been looking into UMPV and had some questions.

1.) I was looking how to “replace” open items instead of add to playlist (append)
I came across a user post here that mentioned that they replaced “append” with “replace” to achieve this in the following line of code
sock.send(("raw loadfile " + f + " append\n").encode("utf-8"))
I tried this and it worked.
My question is, how did user know to change append / replace
Is there a UMPV manual / readme file that mentions this anywhere?
Just wondering what else might be possible with this script.

2.) I like how this works by default, but I was wondering if the following is possible.
-Say you have your MPV set to open in fullscreen
-Then using your window managers hotkey you toggle to a different application, which in this case is a file manager.
-While MPV is running but not in focus, you navigate to another media file to play it and hit enter to open it
-Because UMPV is running MPV does not open up an new instance, but uses the current one.
However, in doing this the media begins to run but MPV does not take the focus back bringing it to the “front” (on top of file manager) in fullscreen.

Again, I like the above behavior when opening audio files because then I can stay in file manager, so not that important.
However I think I would like to be able when opening video files for MPV window to take the focus back and be playing the video file in fullscreen again.
Does anyone know if anything like this is possible?

Thank you to anyone who reads this and for any info shared

**Comment by guidocella** [comment]
Commands are documented at https://mpv.io/manual/master/#command-interface-[%3Coptions%3E]]]

mpv doesn't have a feature to gain focus so you need to use `xdotool` or `wmctrl`.

**Comment by ahaoboy** [comment]
mpv has an optop attribute, which is similar to focus

**Comment by advice2020** [comment]
Thank you for your response I had looked this up a while back when I read this but didn't see anything about this.
Can someone describe where to find this "optop" attribute?
Thank You

**Reply by guidocella** [reply]
https://mpv.io/manual/master/#options-ontop

**Reply by advice2020** [reply]
Oh "ontop" other user mentioned "optop" that would be why my keyword search did not generate anything in the manual :)
Thank You
